scrape_books/spiders/books.py

```diff
@@ -0,0 +1,49 @@
+import re
+
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/catalogue/category/books_1/page-1.html"]
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css(".product_pod"):
+            book_url = response.urljoin(book.css("h3 > a::attr(href)").get())
+            yield scrapy.Request(book_url, callback=self._parse_book_from_detail_page)
+
+        next_page = response.css("ul > li.next > a::attr(href)").get()
+        if next_page:
+            next_page = response.urljoin(next_page)
+            yield scrapy.Request(next_page, callback=self.parse)
+
+    def _parse_book_from_detail_page(self, response: Response) -> dict:
+        title = response.css("h1::text").get()
+        price = float(response.css("p.price_color::text").get().replace("£", ""))
+
+        availability_text = response.css("p.instock.availability").get()
+        amount_in_stock = re.search(r"\d+", availability_text).group()
+
+        rating_class = response.css("p.star-rating::attr(class)").get()
+        rating = rating_class.split()[-1]
+        rating_value = {
+            "One": 1,
+            "Two": 2,
+            "Three": 3,
+            "Four": 4,
+            "Five": 5
+        }.get(rating, 0)
+
+        category = response.css("ul > li:nth-child(3) > a::text").get()
```

Comment:
Ensure that the category selector correctly targets the desired element. If the structure of the page changes, this might not return the expected result.